type-definitions/most.d.ts

```diff
@@ -0,0 +1,349 @@
+declare type SeedValue<S, V> = { seed: S, value: V };
+declare type TimeValue<V>    = { time: number, value: V };
+
+declare interface Thenable<A> {
+    then<U>(onFulfilled?: (value: A) => U | Thenable<U>, onRejected?: (error: any) => U | Thenable<U>): Thenable<U>;
+    then<U>(onFulfilled?: (value: A) => U | Thenable<U>, onRejected?: (error: any) => void): Thenable<U>;
+    catch<U>(onRejected?: (error: any) => U | Thenable<U>): Thenable<U>;
+}
+
+declare interface Promise<A> {
+  constructor(callback: (resolve: (value?: A | Thenable<A>) => void, reject: (error?: any) => void) => void): Promise<A>;
+
+  then<U>(onFulfilled?: (value: A) => U | Thenable<U>, onRejected?: (error: any) => U | Thenable<U>): Promise<U>;
+  then<U>(onFulfilled?: (value: A) => U | Thenable<U>, onRejected?: (error: any) => void): Promise<U>;
+
+  catch<U>(onRejected?: (error: any) => U | Thenable<U>): Promise<U>;
+}
+
+declare interface Generator<A, B, C> {}
+declare interface Iterable<A> {}
+
+declare type CreateGenerator<A> = (...args: Array<any>) => Generator<A|Promise<A>, any, any>;
+
+
+export interface Stream<A> {
+  reduce<B>(f: (b: B, a: A) => B, b: B): Promise<B>;
+  observe(f: (a: A) => any): Promise<any>;
+  forEach(f: (a: A) => any): Promise<any>;
+  drain(): Promise<any>;
+
+  constant<B>(b: B): Stream<B>;
+  map<B>(f: (a: A) => B): Stream<B>;
+  tap(f: (a: A) => any): Stream<A>;
+  chain<B>(f: (a: A) => Stream<B>): Stream<B>;
+  flatMap<B>(f: (a: A) => Stream<B>): Stream<B>;
+
+  ap<B, C>(fs: Stream<(a: A) => B>): Stream<C>;
+
+  // Note: Without higher-kinded types, the types for these
+  // cannot be written in a completely safe manner; See https://github.com/Microsoft/TypeScript/issues/1290
+  // For better type safety, consider using most.join/switch/switchLatest with thru
+  join(): A;
+  switch(): A;
+  switchLatest(): A;
+
+  continueWith(f: (a: any) => Stream<A>): Stream<A>;
+  concatMap<B>(f: (a: A) => Stream<B>): Stream<B>;
+  mergeConcurrently<B>(concurrency: number): Stream<B>;
+  merge(...ss: Array<Stream<A>>): Stream<A>;
+  mergeArray(streams: Array<Stream<A>>): Stream<A>;
+
+  combine<B, R>(
+    fn: (a: A, b: B) => R,
+    b: Stream<B>
+  ): Stream<R>;
+  combine<B, C, R>(
+    fn: (a: A, b: B, c: C) => R,
+    b: Stream<B>,
+    c: Stream<C>
+  ): Stream<R>;
+  combine<B, C, D, R>(
+    fn: (a: A, b: B, c: C, d: D) => R,
+    b: Stream<B>,
+    c: Stream<C>,
+    d: Stream<D>
+  ): Stream<R>;
+  combine<B, C, D, E, R>(
+    fn: (a: A, b: B, c: C, d: D, e: E) => R,
+    b: Stream<B>,
+    c: Stream<C>,
+    d: Stream<D>,
+    e: Stream<E>
+  ): Stream<R>;
+
+  combineArray<B, R>(
+    fn: (a: A, b: B) => R,
+    array: [Stream<B>]
```

Comment:
Not a blocker for merging, but it's not clear to me what to name this parameter.  For example, it's obviously an array from the type signature, so the name `array` is redundant.  I was going to suggest the name `streams`, but that's *also* redundant since it's obviously an array of Stream.

Struggling to find a good name ... `inputs`??